config/main.py

```diff
@@ -3153,111 +3214,110 @@ def del_vrf_vni_map(ctx, vrfname):
-    pass
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    ctx.obj = {}
+    ctx.obj['config_db'] = config_db
 
 @route.command('add', context_settings={"ignore_unknown_options":True})
 @click.argument('command_str', metavar='prefix [vrf <vrf_name>] <A.B.C.D/M> nexthop <[vrf <vrf_name>] <A.B.C.D>>|<dev <dev_name>>', nargs=-1, type=click.Path())
 @click.pass_context
 def add_route(ctx, command_str):
     """Add route command"""
-    if len(command_str) < 4 or len(command_str) > 9:
-        ctx.fail("argument is not in pattern prefix [vrf <vrf_name>] <A.B.C.D/M> nexthop <[vrf <vrf_name>] <A.B.C.D>>|<dev <dev_name>>!")
-    if "prefix" not in command_str:
-        ctx.fail("argument is incomplete, prefix not found!")
-    if "nexthop" not in command_str:
-        ctx.fail("argument is incomplete, nexthop not found!")
-    for i in range(0, len(command_str)):
-        if "nexthop" == command_str[i]:
-            prefix_str = command_str[:i]
-            nexthop_str = command_str[i:]
-    vrf_name = ""
-    cmd = 'sudo vtysh -c "configure terminal" -c "ip route'
-    if prefix_str:
-        if len(prefix_str) == 2:
-            prefix_mask = prefix_str[1]
-            cmd += ' {}'.format(prefix_mask)
-        elif len(prefix_str) == 4:
-            vrf_name = prefix_str[2]
-            prefix_mask = prefix_str[3]
-            cmd += ' {}'.format(prefix_mask)
-        else:
-            ctx.fail("prefix is not in pattern!")
-    if nexthop_str:
-        if len(nexthop_str) == 2:
-            ip = nexthop_str[1]
-            if vrf_name == "":
-                cmd += ' {}'.format(ip)
-            else:
-                cmd += ' {} vrf {}'.format(ip, vrf_name)
-        elif len(nexthop_str) == 3:
-            dev_name = nexthop_str[2]
-            if vrf_name == "":
-                cmd += ' {}'.format(dev_name)
-            else:
-                cmd += ' {} vrf {}'.format(dev_name, vrf_name)
-        elif len(nexthop_str) == 4:
-            vrf_name_dst = nexthop_str[2]
-            ip = nexthop_str[3]
-            if vrf_name == "":
-                cmd += ' {} nexthop-vrf {}'.format(ip, vrf_name_dst)
+    config_db = ctx.obj['config_db']
+    key, route = cli_sroute_to_config(ctx, command_str)
+
+    # If defined intf name, check if it exists
+    if 'ifname' in route:
+        if (not route['ifname'] in config_db.get_keys('VLAN') and
+            not route['ifname'] in config_db.get_keys('INTERFACE') and
+            not route['ifname'] in config_db.get_keys('PORTCHANNEL')):
+            ctx.fail('interface {} doesn`t exist'.format(route['ifname']))
+
+    # Check if exist entry with key
+    keys = config_db.get_keys('STATIC_ROUTE')
+    if key in keys:
+        # If exist update current entry
+        current_entry = config_db.get_entry('STATIC_ROUTE', key)
+
+        for entry in ['nexthop', 'nexthop-vrf', 'ifname']:
+            if not entry in current_entry:
+                    current_entry[entry] = ''
```

Comment:
Please remove extra spaces to make the indentation consistent.

Reply:
Done